include/hax_interface.h

```diff
@@ -224,6 +225,15 @@ struct hax_set_ram_info {
     uint64_t va;
 } PACKED;
 
+struct hax_set_ram_info2 {
+    uint64_t pa_start;
+    uint64_t size;
+    uint8_t flags;
+    uint8_t pad[7];
```

Comment:
I think that's good idea. I simply copied the old ioctl data struct without thinking about more. I will update a new patch.
I have a QEMU side patch too. I can share with you if you need to test. It is not ready to send out since it is written for simple testing.

Reply:
Cool, a simple patch will do. Please feel free to send it to me after you post the next update.